nexus/types/src/deployment.rs

```diff
@@ -1000,6 +1043,193 @@ impl fmt::Display for BlueprintZoneImageVersion {
     }
 }
 
+#[derive(
+    Clone, Debug, Eq, PartialEq, JsonSchema, Deserialize, Serialize, Diffable,
+)]
+pub struct PendingMgsUpdates {
+    // Only one outstanding MGS-managed update is allowed for a given baseboard.
+    by_baseboard: BTreeMap<Arc<BaseboardId>, PendingMgsUpdate>,
```

Comment:
I think this could use `IdMap` instead, to guarantee there's no mismatch between the key and the value's `baseboard_id`?

Reply:
Good call -- I'll try that.  When I initially did this, I tried that, but `IdMap` required the key to be `Copy`.  It looks like it'll be a better fit now.

Reply:
Done in 244c42b4b2c750f73e355a614b00e64e3f126874.